File: CppPractice/Project1/Project1/Leetcode452.cpp
#include <string>

#include <vector>

#include <iostream>

#include <stdlib.h>

#include <time.h>

#include <sstream>

#include <algorithm>

#include <map>

#include <bitset>

using namespace std;

class Solution {
public:
    int findMinArrowShots(vector<vector<int>>& points) {
        if (points.size()==0)
        {
            return 0;
        }
        int ans = 1;
        sort(points.begin(), points.end(), [](vector<int> a, vector<int> b) {return a[1] < b[1]; });
        int anchor = points[0][1];
        for (int i = 1; i < points.size(); i++)
        {
            if (points[i][0]>anchor)
            {
                anchor = points[i][1];
                ans++;
            }
        }
        return ans;
    }
};

File: CSharpPractice/LeetcodePractice/Leetcode452.cs
public class Solution452 {
    public int FindMinArrowShots(int[][] points) 
    {
        Array.Sort(points,(a,b)=>a[1].CompareTo(b[1]));
        int ans=1;
        int end= points[0][1];
        if(points.Length<1){return 0;}
        for(int i=1;i<points.Length;i++)
        {
            if(points[i][0]>end)
            {
                end=points[i][1];
                ans++;
            }
        }
        return ans;
    }
}